spring-cloud-stream-binder-kafka-docs/src/main/asciidoc/overview.adoc

```diff
@@ -429,4 +430,69 @@ On the other hand, if auto topic creation is disabled on the server, then care m
+For more information on Kafka Streams, see https://kafka.apache.org/0110/documentation/streams/developer-guide[Kafka Streams API Developer Manual]
+
+Here are the maven coordinates for the Spring Cloud Stream KStream binder artifact.
+
+[source,xml]
+----
+<dependency>
+  <groupId>org.springframework.cloud</groupId>
+  <artifactId>spring-cloud-stream-binder-kstream</artifactId>
+</dependency>
+----
+
+In addition to leveraging the Spring Cloud Stream programming model which is based on Spring Boot, one of the main other benefits that the Kstream binder provides is the fact that it avoids the boilerplate configuration that one needs to write when using the Kafka Streams API directly.
+High level streams DSL provided through the Kafka Streams API can be used through Spring Cloud Stream in the current support.
+
+=== Usage example of high level streams DSL
+
+This application will listen from a Kafka topic and write the word count for each unique word that it sees in a 5 seconds time window.
+
+[source]
+----
+@SpringBootApplication
+@EnableBinding(KStreamProcessor.class)
+public class WordCountProcessorApplication {
+
+	@StreamListener("input")
+	@SendTo("output")
+	public KStream<?, String> process(KStream<?, String> input) {
+		return input
+				.flatMapValues(value -> Arrays.asList(value.toLowerCase().split("\\W+")))
+				.map((key, word) -> new KeyValue<>(word, word))
+				.groupByKey(Serdes.String(), Serdes.String())
+				.count(TimeWindows.of(5000), "store-name")
+				.toStream()
+				.map((w, c) -> new KeyValue<>(null, "Count for " + w.key() + ": " + c));
+	}
+
+	public static void main(String[] args) {
+		SpringApplication.run(WordCountProcessorApplication.class, args);
+	}
+----
+
+If you build it as spring boot runnable fat jar, you can run the above example in the following way:
+
+[source]
+----
+java -jar uber.jar  --spring.cloud.stream.bindings.input.destination=words --spring.cloud.stream.bindings.output.destination=counts
+----
+
+This means that the application will listen from the incoming Kafka topic words and write to the output topic counts.
+
+Spring Cloud Stream will ensure that the messages from both the incoming and outgoing topics are bound as KStream objects.
+As one may observe, the developer can exclusively focus on the business aspects of the code, i.e writing the logic required in the processor rather than setting up the streams specific configuration required by the Kafka Streams infrastructure.
+All those boilerplate is handled by Spring Cloud Stream behind the scenes.
+
+=== Kafka Streams properties
+
+The properties must be prefixed with `spring.cloud.stream.kstream.binder.`.
+
+streamConfiguration::
+  Map with a key/value pair containing properties pertaining to Kafka Streams API.
+  For all the properties that may go into streams configuration, see http://www.trieuvan.com/apache/kafka/0.10.2.1/javadoc/org/apache/kafka/streams/StreamsConfig.html[StreamsConfig javadoc].
```

Comment:
Yeah... Unfortunately we can't have this without version. 
How about just say: 

> see `StreamsConfig` JavaDocs for more information

?